### PR TITLE
Mv expiry fallback

### PR DESCRIPTION
**Pull request by matthewvon**
Create a release without expiry support that can process expiry enabled files.  This is to be a fallback release should any portion of Riak fail and require a fallback to a prior release.

### CODE REVIEW

File: db/dbformat.h [hunk 1]
```diff
@@ -200,8 +245,12 @@ inline bool ParseInternalKey(const Slice& internal_key,
   unsigned char c = num & 0xff;
   result->sequence = num >> 8;
   result->type = static_cast<ValueType>(c);
-  result->user_key = Slice(internal_key.data(), n - 8);
-  return (c <= static_cast<unsigned char>(kTypeValue));
+  if (IsExpiryKey((ValueType)c))
+    result->expiry=DecodeFixed64(internal_key.data() + n - 16);
```
Comment (paulplace) [inline db/dbformat.h]: Would be good to get rid of hard-coded "- 16", much like most of the "- 8"s were replaced.

Comment (matthewvon) [inline db/dbformat.h]: Done.

File: db/dbformat.h [hunk 2]
```diff
@@ -107,18 +111,59 @@ extern void AppendInternalKey(std::string* result,
 extern bool ParseInternalKey(const Slice& internal_key,
                              ParsedInternalKey* result);
 
+inline ValueType ExtractValueType(const Slice& internal_key) {
+  assert(internal_key.size() >= 8);
+  const size_t n = internal_key.size();
+  unsigned char c = DecodeLeastFixed64(internal_key.data() + n - 8);
```
Comment (paulplace) [inline db/dbformat.h]: Can this hardcoded "- 8" be replaced with something more descriptive?

Comment (matthewvon) [inline db/dbformat.h]: done

**Comment by paulplace** [comment]
+1 c56a2b6

**Pull request merged**